py-polars/src/functions/lazy.rs

```diff
@@ -435,7 +435,10 @@ pub fn lit(value: &Bound<'_, PyAny>, allow_object: bool) -> PyResult<PyExpr> {
         Ok(dsl::lit(Null {}).into())
     } else if let Ok(value) = value.downcast::<PyBytes>() {
         Ok(dsl::lit(value.as_bytes()).into())
-    } else if value.get_type().qualname().unwrap() == "Decimal" {
+    } else if matches!(
+        value.get_type().qualname().unwrap().as_str(),
+        "date" | "datetime" | "Decimal"
```

Comment:
FYI will make a quick followup PR to add `time` and `timedelta`.

Reply:
Nice; in the meantime let's get this merged before I forget again! 🤓